src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Unix.cs

```diff
@@ -0,0 +1,149 @@
+            FileStreamHelpers.CheckFileCall(result, path);
+            return status.Size;
+        }
+
+        private static unsafe int ReadAtOffset(SafeFileHandle handle, Span<byte> buffer, long fileOffset)
+        {
+            fixed (byte* bufPtr = &MemoryMarshal.GetReference(buffer))
+            {
+                int result = Interop.Sys.PRead(handle, bufPtr, buffer.Length, fileOffset);
+                FileStreamHelpers.CheckFileCall(result, path: null);
+                return result;
+            }
+        }
+
+        private static unsafe long ReadScatterAtOffset(SafeFileHandle handle, IReadOnlyList<Memory<byte>> buffers, long fileOffset)
+        {
+            MemoryHandle[] handles = new MemoryHandle[buffers.Count];
+            Span<Interop.Sys.IOVector> vectors = buffers.Count <= IovStackThreshold ? stackalloc Interop.Sys.IOVector[IovStackThreshold] : new Interop.Sys.IOVector[buffers.Count ];
+
+            for (int i = 0; i < buffers.Count; i++)
+            {
+                Memory<byte> buffer = buffers[i];
+                MemoryHandle memoryHandle = buffer.Pin();
+                vectors[i] = new Interop.Sys.IOVector { Base = (byte*)memoryHandle.Pointer, Count = (UIntPtr)buffer.Length };
+                handles[i] = memoryHandle;
+            }
+
+            long result;
+            fixed (Interop.Sys.IOVector* pinnedVectors = &MemoryMarshal.GetReference(vectors))
+            {
+                result = Interop.Sys.PReadV(handle, pinnedVectors, buffers.Count, fileOffset);
+            }
+
+            foreach (MemoryHandle memoryHandle in handles)
+            {
+                memoryHandle.Dispose();
+            }
+
+            return FileStreamHelpers.CheckFileCall(result, path: null);
+        }
+
+        private static ValueTask<int> ReadAtOffsetAsync(SafeFileHandle handle, Memory<byte> buffer, long fileOffset,
+            CancellationToken cancellationToken)
+        {
+            return new ValueTask<int>(
+                new TaskFactory<int>(
+                        cancellationToken,
+                        TaskCreationOptions.None,
+                        TaskContinuationOptions.DenyChildAttach,
+                        TaskScheduler.Default)
+                    .StartNew(() => ReadAtOffset(handle, buffer.Span, fileOffset), cancellationToken));
+        }
+
+        private static ValueTask<long> ReadScatterAtOffsetAsync(SafeFileHandle handle, IReadOnlyList<Memory<byte>> buffers,
+            long fileOffset, CancellationToken cancellationToken)
+        {
+            return new ValueTask<long>(
+                new TaskFactory<long>(
+                        cancellationToken,
+                        TaskCreationOptions.None,
+                        TaskContinuationOptions.DenyChildAttach,
+                        TaskScheduler.Default)
+                    .StartNew(() => ReadScatterAtOffset(handle, buffers, fileOffset), cancellationToken));
```

Comment:
Same.  There's no need to construct a TaskFactory, and we can box a single tuple rather than allocate both a closure and a delegate for the capture.